src/netbase.cpp

```diff
@@ -42,6 +42,53 @@ bool fNameLookup = DEFAULT_NAME_LOOKUP;
 int g_socks5_recv_timeout = 20 * 1000;
 static std::atomic<bool> interruptSocks5Recv(false);
 
+std::vector<CNetAddr> WrappedGetAddrInfo(const std::string& name, bool allow_lookup)
+{
+    struct addrinfo aiHint;
+    memset(&aiHint, 0, sizeof(struct addrinfo));
+
+    // We want a TCP port, which is a streaming socket type
+    aiHint.ai_socktype = SOCK_STREAM;
+    aiHint.ai_protocol = IPPROTO_TCP;
+    // We don't care which address family (IPv4 or IPv6) is returned
+    aiHint.ai_family = AF_UNSPEC;
+    // If we allow lookups of hostnames, use the AI_ADDRCONFIG flag to only
+    // return addresses whose family we have an address configured for.
+    //
+    // If we don't allow lookups, then use the AI_NUMERICHOST flag for
+    // getaddrinfo to only decode numerical network addresses and suppress
+    // hostname lookups.
+    aiHint.ai_flags = allow_lookup ? AI_ADDRCONFIG : AI_NUMERICHOST;
+
+    struct addrinfo* aiRes = nullptr;
+    int nErr = getaddrinfo(name.c_str(), nullptr, &aiHint, &aiRes);
+    if (nErr) {
+        return {};
+    }
+
+    // Traverse the linked list starting with aiTrav, add all non-internal
+    // IPv4,v6 addresses to vIP while respecting nMaxSolutions.
+    struct addrinfo* aiTrav = aiRes;
+    std::vector<CNetAddr> resolved_addresses;
+    while (aiTrav != nullptr) {
+        if (aiTrav->ai_family == AF_INET) {
+            assert(aiTrav->ai_addrlen >= sizeof(sockaddr_in));
+            resolved_addresses.emplace_back(((struct sockaddr_in*)(aiTrav->ai_addr))->sin_addr);
+        }
+        if (aiTrav->ai_family == AF_INET6) {
+            assert(aiTrav->ai_addrlen >= sizeof(sockaddr_in6));
+            struct sockaddr_in6* s6 = (struct sockaddr_in6*)aiTrav->ai_addr;
```

Comment:
here and line 76, can we use [named casts](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts-named)?

Reply:
I like named cast but they're not applicable in this case.

Reply:
Sure. Now using `reinterpret_cast`.